rs/p2p/quic_transport/src/connection_handle.rs

```diff
@@ -50,6 +54,64 @@ pub struct ConnectionHandle {
     conn_id: ConnId,
 }
 
+fn observe_conn_error(err: &ConnectionError, op: &str, metrics: &QuicTransportMetrics) {
+    match err {
+        // TODO: most likely this can be made infallible
+        ConnectionError::LocallyClosed => metrics
+            .connection_handle_errors_total
+            .with_label_values(&[op, ERROR_LOCALLY_CLOSED_CONN])
+            .inc(),
+        ConnectionError::ApplicationClosed(_) => metrics
+            .connection_handle_errors_total
+            .with_label_values(&[op, ERROR_APP_CLOSED_CONN])
+            .inc(),
+        _ => metrics
+            .connection_handle_errors_total
+            .with_label_values(&[op, ERROR_INTERNALLY_CLOSED_CONN])
+            .inc(),
+    }
+}
+
+fn observe_write_error(err: &WriteError, op: &str, metrics: &QuicTransportMetrics) {
```

Comment:
From the error metrics you saw in production. Do you think it's worthwhile to include a label for if the write error comes from the Rpc handler or  the Rpc caller?